core/trino-spi/src/main/java/io/trino/spi/connector/Constraint.java

```diff
@@ -69,7 +91,15 @@ private Constraint(TupleDomain<ColumnHandle> summary, Optional<Predicate<Map<Col
     }
 
     /**
-     * A predicate that can be used to filter data. If present, it is equivalent to, or stricter than, {@link #getSummary()}.
+     * @return an expression predicate which is different from, and should be AND-ed with, {@link #getSummary} or {@link #predicate} (if present).
+     */
+    public ConnectorExpression getExpression()
```

Comment:
Thanks. Can you please add similar comment to `getSummary` too?

Reply:
Done